### PR TITLE
[Ethflow] using the newest contract

### PR DESCRIPTION
**Pull request by josojo**
Using the newest contracts. I needed to update ethcontract, due to https://github.com/gnosis/ethcontract-rs/issues/854

### CODE REVIEW

File: crates/autopilot/src/database/onchain_order_events.rs [hunk 1]
```diff
@@ -133,8 +133,15 @@ impl<T: Sync + Send + Clone, W: Sync + Send + Clone> EventStoring<ContractEvent>
         events: Vec<EthContractEvent<ContractEvent>>,
         range: RangeInclusive<u64>,
     ) -> Result<()> {
-        let (custom_onchain_data, quotes, broadcasted_order_data, orders) =
-            self.extract_custom_and_general_order_data(events).await?;
+        let order_placement_events = events
+            .into_iter()
+            .filter(|EthContractEvent { data, .. }| {
```
Comment (josojo) [inline crates/autopilot/src/database/onchain_order_events.rs:138]: This is only temporary. THe next PR will also filter for the new cancellation event

File: crates/contracts/build.rs [hunk 2]
```diff
@@ -19,16 +19,7 @@ fn main() {
     println!("cargo:rerun-if-changed=build.rs");
 
     generate_contract_with_config("CoWSwapEthFlow", |builder| {
-        builder
-            .contract_mod_override("cowswap_eth_flow")
```
Comment (josojo) [inline crates/contracts/build.rs:23]: We decided to deploy 1 contract for staging and 1 for prod. Hence, I don't import contract address as before, but rather via env variables

File: crates/autopilot/src/arguments.rs [hunk 3]
```diff
@@ -23,7 +23,7 @@ pub struct Arguments {
     pub price_estimation: price_estimation::Arguments,
 
     /// Address of the ethflow contract
-    #[clap(long, env, default_value = "31172bb2b5f97e8e89cf3376495d7bc7252f5a53")]
+    #[clap(long, env, default_value = "26c3801b4840dab317cedfd7aeaf9f45fdd22830")]
```
Comment (nlordell) [inline crates/autopilot/src/arguments.rs:26]: ubernit: 
```suggestion
    #[clap(long, env, default_value = "0x26c3801b4840DaB317CedFd7aeAf9f45Fdd22830")]
```

File: crates/refunder/src/arguments.rs [hunk 4]
```diff
@@ -35,6 +36,10 @@ pub struct Arguments {
     /// The Ethereum node URL to connect to.
     #[clap(long, env, default_value = "http://localhost:8545")]
     pub node_url: Url,
+
+    /// Address of the ethflow contract
+    #[clap(long, env, default_value = "26c3801b4840dab317cedfd7aeaf9f45fdd22830")]
```
Comment (nlordell) [inline crates/refunder/src/arguments.rs:41]: ubernit:
```suggestion
    #[clap(long, env, default_value = "0x26c3801b4840DaB317CedFd7aeAf9f45Fdd22830")]
```

**Review by nlordell — APPROVED** [review]
LGTM - just a tiny ubernit, but fine with or without it.

File: crates/refunder/src/arguments.rs [hunk 5]
```diff
@@ -45,6 +54,7 @@ impl std::fmt::Display for Arguments {
         writeln!(f, "min_slippage_bps: {}", self.min_slippage_bps)?;
         writeln!(f, "db_url: SECRET")?;
         writeln!(f, "node_url: {}", self.node_url)?;
+        writeln!(f, "ethflow_contract: {}", self.ethflow_contract)?;
```
Comment (MartinquaXD) [inline crates/refunder/src/arguments.rs:57]:
```suggestion
        writeln!(f, "ethflow_contract: {:?}", self.ethflow_contract)?;
```
With `Display` formatting: `ethflow contract: 0x26c3…2830`
With `Debug` formatting: `ethflow contract: 0x26c3801b4840dab317cedfd7aeaf9f45fdd22830`
I suggest the same change for the `autopilot` crate as well.

**Review by MartinquaXD — APPROVED** [review]

**Review by vkgnosis — APPROVED** [review]

**Pull request merged**